src/components/oculus-touch-controls.js

```diff
@@ -320,20 +357,130 @@ module.exports.Component = registerComponent('oculus-touch-controls', {
     });
   },
 
+  applyOffset: function (model) {
+    model.position.copy(this.displayModel[this.data.hand].modelPivotOffset);
+    model.rotation.copy(this.displayModel[this.data.hand].modelPivotRotation);
+  },
+
+  multiMeshFix: function (button) {
+    if (!this.buttonMeshes[button]) return;
+    this.buttonMeshes[button].traverse((node) => {
+      if (node.type !== 'Mesh') return;
+      var originalMaterial = node.material;
+      this.buttonMeshes[button].naturalColor = originalMaterial.color;
+      node.material = new THREE.MeshStandardMaterial();
+
+      // preserve details; not iterable, so we have to use Object.keys()
```

Comment:
Why are you saying it's not iterable? What did you try?
You gave me some doubt here so I add to verify again my own code. :D 
I have other code that does this:
```
for (const propertyName in material) {
    const texture = material[propertyName];
    if (texture instanceof THREE.Texture) {
      const image = texture.source.data;
      if (image instanceof ImageBitmap) {
        image.close && image.close();
      }
      texture.dispose();
    }
  }
```
It's definitely enumerable.

Reply:
You're right. I'm not sure how that happened, but my first attempt using for-in didn't sreem to show the keys I needed, so I switched to this and it worked. It may have just been a quick caching issue and a resulting assumption? But retrying this, you're definitely correct.

That said, for-in loops are ES6... a bit confusing, since for-in is recommended as a best practice in the aframe docs if I'm not mistaken, as it avoids an unnecessary array generation? Not terribly relevant here since this runs once, of course.

And all that said, seems the copy method you suggest will be the best option. This iteration worked, but I'm definitely glad to find a more efficient way to do this! I was riffing on an example I found elsewhere--I tried to make it more efficient and fit to my example, then things broke, then I fixed the things that broke. 

While I think this will be slightly less efficient, it is cleaner code, and also should 'fix' any other 'links' between the child materials--not relevant right now, but makes this more future proof and makes the code a better example.

Reply:
for-in is ES5, for-of is ES6.
https://hacks.mozilla.org/2015/04/es6-in-depth-iterators-and-the-for-of-loop/
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Aframe actually use some ES6 features in some places and those changes were actually introduced by @dmarcos ;-). You can easily see where because the minify we use doesn't support ES6 syntax. I created #5106 for this.